Style ledger notifications like update notifications

ayumi opened the pull request:
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Partially implements #3257

- Extract/refactor updateBar HTML and CSS for use in notificationBar
- Add custom styling of notification bar and buttons

Auditors: @bradleyrichter @diracdeltas 

Test Plan:
1. Change conditional in https://github.com/brave/browser-laptop/blob/feature/ledger-notifications-update-style/app/ledger.js#L1220 to `true`
2. Observe the funding notification is styled like the update bar

Preview:
<img width="1318" alt="screen shot 2016-09-09 at 17 26 40" src="https://cloud.githubusercontent.com/assets/521861/18406613/27172368-76b4-11e6-80bb-6b1a750a6f03.png">

bradleyrichter commented:
sweet! thanks!

diracdeltas commented:
i haven't looked into why, but this appears to be applying the style of the first notification bar to all the notification bars below it, regardless of whether they are ledger notifications.

to repro:
1. go to https://www.browserleaks.com/geo, see the geolocation notification bar appear
2. close browser, edit https://github.com/brave/browser-laptop/blob/feature/ledger-notifications-update-style/app/ledger.js#L1220 to true
3. re-open browser
4. observe that geolocation notification is styled like ledger notifications now
![screen shot 2016-09-09 at 6 14 22 pm](https://cloud.githubusercontent.com/assets/549654/18406948/a5b34472-76b9-11e6-88f6-ead662628ab9.png)

if you close the first notification so that the geolocation notification is on top, then the ledger notification loses its style
![screen shot 2016-09-09 at 6 15 05 pm](https://cloud.githubusercontent.com/assets/549654/18406953/c7c8447c-76b9-11e6-85a2-e9aefc636b5f.png)

otherwise i feel it

ayumi commented:
I initially tried to style each notification separately, but that looked really ugly due to variations in margin/padding.
So I decided it'd be an adequate fallback to style them all like the first notification.

Alternatively we could sort and group notifications by style, and separate them with borders on top/bottom.

ayumi commented:
I removed the style override based on the 0-th notification, and added borders. What do you think?

<img width="852" alt="screen shot 2016-09-09 at 19 39 28" src="https://cloud.githubusercontent.com/assets/521861/18407450/8b84d3a2-76c5-11e6-90c7-cc187249efd3.png">

diracdeltas commented:
i feel it even more now

bradleyrichter commented:
We need to sync up the style on the yellow alert. Can we use 3 buttons instead?

No / yes / always 

Or similar....?




> On Sep 9, 2016, at 8:16 PM, yan zhu (@bcrypt) <notifications@github.com> wrote:
> 
> i feel it even more now
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

diracdeltas commented:
> We need to sync up the style on the yellow alert. Can we use 3 buttons instead?

It would have to be four buttons (No, Yes, Always No, Always Yes)

ayumi commented:
@diracdeltas Notifications are now sorted by style and prioritize styled notifications over the defaults:
<img width="969" alt="screen shot 2016-09-09 at 23 18 18" src="https://cloud.githubusercontent.com/assets/521861/18408462/52a5754e-76e5-11e6-9eb6-d8571b7df020.png">

@bradleyrichter Besides restyling the 2 ledger notifications, this doesn't affect other notifications so I think it's out of scope for this PR.

bsclifton commented on js/stores/appStore.js:
```diff
@@ -507,13 +507,34 @@ const handleAppAction = (action) => {
+
+      // Insert notification next to those with the same style, or at the end
+      let insertIndex = notifications.size
+      const style = action.detail.get
```
for checking against immutable, you might check for the presence of toJS instead; I know a lot of other code does that (just a consistency nitpick)

bsclifton commented on js/stores/appStore.js:
```diff
@@ -507,13 +507,34 @@ const handleAppAction = (action) => {
+        } else {
+          // Insert after the last notification with a style
+          insertIndex = notifications.findLastIndex((notification) => {
+            return notification.get('options').get('style')
```
to be explicit, you could coerce this to a bool, like:
`return !!notification.get('options').get('style')`

But usually when something has that "smell", it's best to just check the type:
`return typeof notification.get('options').get('style') === 'string'`

ayumi replied:
Thanks! I'm checking that it's a string now.

bsclifton commented:
Comments left on the last commit :+1:

diracdeltas commented on js/stores/appStore.js:
```diff
@@ -505,13 +507,34 @@ const handleAppAction = (action) => {
+      // Insert notification next to those with the same style, or at the end
+      let insertIndex = notifications.size
+      const style = action.detail.get
+        ? action.detail.get('options').get('style')
```
for future reference, you can just do `thing.getIn(['options', 'style'])` to avoid it throwing an error if `thing.get('options')` is falsy

diracdeltas commented:
this seems to cause some of the notification bar tests to fail. https://travis-ci.org/brave/browser-laptop/builds/159031606

otherwise lgtm, feel free to merge!

ayumi commented:
@diracdeltas thx, i've updated navigationBar test to use the new css selector
